src/Symfony/Component/Workflow/Workflow.php

```diff
@@ -119,7 +120,15 @@ public function buildTransitionBlockerList($subject, string $transitionName): Tr
             $transitionBlockerList = $this->buildTransitionBlockerListForTransition($subject, $marking, $transition);
 
             if ($transitionBlockerList->isEmpty()) {
-                continue;
+                return $transitionBlockerList;
+            }
+
+            // We prefer to return transitions blocker by something else than
+            // marking. Because it means the marking was OK Transition are
+            // deterministic : it's not possible to have many transition enabled
+            // at the same time that match the same marking with the same name
+            if (!$transitionBlockerList->has(TransitionBlocker::BLOCKED_BY_MARKING)) {
```

Comment:
But then the comment will be harder to understand because it applies to only one part of the if statement 

I prefer to keep it as it. Are you OK with that?

Reply:
yes, no blocker for me